### DISCUSSION TITLE
Add link annotations from json

### DISCUSSION THREAD

**Discussion by cagilla** [pdfcpu/pdfcpu#712] [Q&A] [answered]
Is it possible to add a NewLinkAnnotation as part of a json file?  I could not find an example.  I would like to create a link from page 1 to page 3.  for instance.

**Comment by hhrutter** [comment, marked as answer]
Not right now, but you can file a feature request issue.

There's a lot a LinkAnnotation can do and how to map that over to JSON without overwhelming users is a tricky part.

Meanwhile you can do it in Go: 

See func TestAddLinkAnnotationWithDest in https://github.com/pdfcpu/pdfcpu/blob/master/pkg/api/test/annotation_test.go

**Reply by cagilla** [reply]
Thank you... at least you did not respond and tell me it was clearly in the help :-)

**Comment by hhrutter** [comment]
Update: as of v0.13.0 this is partially resolved.

URL-style links from generated content are supported, but arbitrary internal destination link annotations via JSON still not addressed.

page/destination links need a separate focused issue.
